### PR TITLE
Moved theme functionality to the redux store

### PR DESCRIPTION
**Pull request by ravisrc**
## Changes made

Moved the theme functionality to the Redux store for a better state management and also a good reliability of front-end

## Issues/PRs related

Related to #2

### CODE REVIEW

**Comment by rashil2000** [comment]
Looks great on first glance, I'll test this tomorrow. 

On a side note, include screenshots only if you have made visual changes to the site. You don't need to include screenshots of code. You should remove them.

**Comment by rashil2000** [comment]
@ravisrc See the deploy preview mentioned the checks section below - here's the link https://deploy-preview-4--dystuss.netlify.app/

Upon first load, it's showing me the dark theme, when clearly it's day time and it should default to a light one.

**Comment by ravisrc** [comment]
Before, the theme is set to "night" if nothing is found in localStorage, but now it has been changed to set automatically as per time.

File: src/store/reducer.js [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+else timeSlot = "Night";
+
+themeDictionary["Auto-change"] = {...themeDictionary[timeSlot]}
+
+// Loading a pre-set theme (if any) from LocalStorage
+
+let presetTheme;
+let themeSet = JSON.parse(localStorage.getItem("themeSet"));
+
+if (themeSet === "Auto-change") {
```
Comment (rashil2000) [inline src/store/reducer.js:72]: I think this entire `if` block can be simplified like this: 
```javascript
if (themeDictionary[themeSet] !== undefined)
  presetTheme = { ...themeDictionary[themeSet] };
else
  presetTheme = { ...themeDictionary[timeSlot] };
```

What do you think?

**Review by rashil2000 — CHANGES_REQUESTED** [review]

File: src/App.js [hunk 2]
```diff
@@ -15,70 +15,27 @@ import About from "./views/About";
...
-        backimg = "/assets/firewatch_7.jpg";
-        break;
-      default:
-        name = "dark";
-        color = "#a0a5ab";
-        backimg = "/assets/firewatch_8.jpg";
+class App extends React.Component {
+
+  shouldComponentUpdate(nextProps, nextState){
+    if(nextProps.theme.color !== this.props.theme.color || nextProps.theme.color === this.props.theme.color ){
```
[the remark this answers is not in the document]
Comment (ravisrc) [reply src/App.js:23]: Logically, I thought of using an expression such that
> App component should update only when `props` except `theme` are changed

There is no need of that now. I am removing it.

File: src/store/reducer.js (hunk 1 above)
Comment (ravisrc) [reply src/store/reducer.js:72]: Yeah absolutely

**Review by rashil2000 — APPROVED** [review]

File: src/App.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (rashil2000) [reply src/App.js]: But how is that condition check helping? `nextProps.theme.color` will either be equal, or unequal to `this.props.theme.color`, right? The condition in that `if` statement will always be true, whether the equality holds or not, don't you think?

**Comment by rashil2000** [comment]
> For API calls and state management, directories separate from 'src' should be made.

Please make this change too, after all the above requested changes have been made.

File: src/App.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (ravisrc) [reply src/App.js]: Yeah Ok. Got it!

**Review by rashil2000 — APPROVED** [review]

**Pull request merged**